### ISSUE TITLE
Inconsistent behavior between backends + missing docs

### ISSUE THREAD

**Issue by elanmart** [fchollet/keras#2155]
The docs for `fit` method of  a `Model` say

> y: labels, as a numpy array.

It's unclear, however, wheter i should be a vector of ints or one-hot matrix.
Perhaps one should mention that there exists `np_utils.to_categorical(y)`?

Also, the following code will fail on Theano but not on Tensorflow backend.

``` python
import numpy as np
import numpy.random as nr

from keras.utils import np_utils
from keras.layers.core import Layer, Dense, Dropout
from keras.layers.normalization import BatchNormalization
from keras.models import Sequential

X = np.random.randn(10000, 500)
y = np.random.randint(0, 64, size=(10000, 1))

test = Sequential()
test.add(Layer(input_shape=(500, )))

test.add(Dense(1024, activation='relu'))
test.add(Dropout(0.5))
test.add(BatchNormalization())

test.add(Dense(64, activation='softmax'))

test.compile('adam', 'categorical_crossentropy')
test.fit(X, y)
```

**Comment by zhangchengcsb** [comment]
Hi @elanmart , if you look up the documentation for categorical_crossentropy, it says:

categorical_crossentropy: Also known as multiclass logloss. Note: using this objective requires that your labels are binary arrays of shape (nb_samples, nb_classes).

So it seems your y should be of shape (10000, 2) and for each row, use [0, 1] and [1, 0] to denote classes.

**Comment by NasenSpray** [comment]
> Also, the following code will fail on Theano but not on Tensorflow backend.

Even nastier, it silently produces a wrong result on Tensorflow.
